client/client.go

```diff
@@ -231,53 +270,189 @@ type (
+	// Deprecated: Replaced by the new worker versioning api.
 	BuildIDOpPromoteSet = internal.BuildIDOpPromoteSet
 
 	// BuildIDOpPromoteIDWithinSet is an operation for UpdateWorkerBuildIdCompatibilityOptions to
 	// promote a BuildID within a set to be the default.
-	// WARNING: Worker versioning is currently experimental
+	// Deprecated: Replaced by the new worker versioning api.
 	BuildIDOpPromoteIDWithinSet = internal.BuildIDOpPromoteIDWithinSet
 
 	// GetWorkerTaskReachabilityOptions is the input to Client.GetWorkerTaskReachability.
-	// WARNING: Worker versioning is currently experimental
+	// Deprecated: Use [DescribeTaskQueueEnhancedOptions] with the new worker versioning api.
 	GetWorkerTaskReachabilityOptions = internal.GetWorkerTaskReachabilityOptions
 
 	// WorkerTaskReachability is the response for Client.GetWorkerTaskReachability.
-	// WARNING: Worker versioning is currently experimental
+	// Deprecated: Replaced by the new worker versioning api.
 	WorkerTaskReachability = internal.WorkerTaskReachability
 
 	// BuildIDReachability describes the reachability of a buildID
-	// WARNING: Worker versioning is currently experimental
+	// Deprecated: Replaced by the new worker versioning api.
 	BuildIDReachability = internal.BuildIDReachability
 
 	// TaskQueueReachability Describes how the Build ID may be reachable from the task queue.
-	// WARNING: Worker versioning is currently experimental
+	// Deprecated: Replaced by the new worker versioning api.
 	TaskQueueReachability = internal.TaskQueueReachability
 
+	// DescribeTaskQueueEnhancedOptions is the input to [Client.DescribeTaskQueueEnhanced].
+	// WARNING: Worker versioning is currently experimental.
+	DescribeTaskQueueEnhancedOptions = internal.DescribeTaskQueueEnhancedOptions
+
+	// TaskQueueVersionSelection is a task queue filter based on versioning.
+	// It is an optional component of [Client.DescribeTaskQueueEnhancedOptions].
+	// WARNING: Worker versioning is currently experimental.
+	TaskQueueVersionSelection = internal.TaskQueueVersionSelection
+
+	// TaskQueueDescription is the response to [Client.DescribeTaskQueueEnhanced].
+	// WARNING: Worker versioning is currently experimental.
+	TaskQueueDescription = internal.TaskQueueDescription
+
+	// TaskQueueVersionInfo includes task queue information per Build ID.
+	// It is part of [Client.TaskQueueDescription].
+	// WARNING: Worker versioning is currently experimental.
+	TaskQueueVersionInfo = internal.TaskQueueVersionInfo
+
+	// TaskQueueTypeInfo specifies task queue information per task type and Build ID.
+	// It is included in [Client.TaskQueueVersionInfo].
+	// WARNING: Worker versioning is currently experimental.
+	TaskQueueTypeInfo = internal.TaskQueueTypeInfo
+
+	// PollerInfo provides information about a worker/client polling a task queue.
+	// It is used by [Client.TaskQueueTypeInfo].
+	// WARNING: Worker versioning is currently experimental.
+	PollerInfo = internal.PollerInfo
```

Comment:
Can this be called `TaskQueuePollerInfo` or `WorkerPollerInfo` or something? I think `PollerInfo` is a bit too generic for this big package.

Reply:
TaskQueuePollerInfo sounds great